### PR TITLE
Complement io.github.endless_sky.endless_sky.appdata.xml

### PR DESCRIPTION
**Pull request by 1Maxnet1**
**Documentation**

## Summary
Based on what is already available in the [LinuxPhoneApps listing](https://linuxphoneapps.org/games/io.github.endless_sky.endless_sky) I added some missing metainfo:
- Adds vcs-browser url

### CODE REVIEW

**Review by TheGiraffe3 — APPROVED** [review]

File: io.github.endless_sky.endless_sky.appdata.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
   <url type="bugtracker">https://github.com/endless-sky/endless-sky/issues</url>
   <url type="help">https://github.com/endless-sky/endless-sky/wiki/PlayersManual</url>
   <url type="homepage">https://endless-sky.github.io/</url>
+  <url type="vcs-browser">https://github.com/endless-sky/endless-sky</url>
```
Comment (warp-core) [inline io.github.endless_sky.endless_sky.appdata.xml:24]: May as well add the link to the actual repository, not just the browser interface for it:
```suggestion
  <url type="vcs-browser">https://github.com/endless-sky/endless-sky</url>
  <url type="vcs-git">https://github.com/endless-sky/endless-sky.git</url>
```

**Comment by tibetiroka** [comment]
Based on the failed CI, it seems that this information is not supported in the appstream file (or maybe not at this location).

>   W: io.github.endless_sky.endless_sky:24: url-invalid-type vcs-browser
  W: io.github.endless_sky.endless_sky:25: url-invalid-type vcs-git

**Comment by 1Maxnet1** [comment]
> https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#spec-component-location
> 
> Could the reason be that we are using a legacy format?

Probably. The check is based on Ubuntu 22.04. which ships with appstream 0.15.4, while the current version is 1.x. So it might not have been included in the older version yet. However I don't think it is too easy to update the hole job to a more modern Ubuntu Base

**Comment by TomGoodIdea** [comment]
this might be relevant: https://github.com/AppImage/appimage.github.io/issues/3243

**Comment by 1Maxnet1** [comment]
> > https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#spec-component-location
> > Could the reason be that we are using a legacy format?
> 
> Probably. The check is based on Ubuntu 22.04. which ships with appstream 0.15.4, while the current version is 1.x. So it might not have been included in the older version yet. However I don't think it is too easy to update the hole job to a more modern Ubuntu Base

As updating the Ubuntu Base of the Continuous Deployment pipeline has some bigger implications. Do you think it might be worth it to run the appimagetool with `--no-appstream` flag to skip the appstream check and place it in the project_checks pipeline? This way it could be run on a recent Ubuntu Base with appstream version 1.x, which should fix the false warnings. I would also be happy to contribute this change.

**Comment by tibetiroka** [comment]
I don't think some extra attributes are worth disabling the checks. The checks are there to tell us when something is wrong, and they performed that function this time. If someone tries to package their own appimage, or run the CD in a fork of the repository, they would need to apply the same workaround, which really shouldn't be necessary.

**Comment by 1Maxnet1** [comment]
> I don't think some extra attributes are worth disabling the checks. The checks are there to tell us when something is wrong, and they performed that function this time. If someone tries to package their own appimage, or run the CD in a fork of the repository, they would need to apply the same workaround, which really shouldn't be necessary.

Sorry, I think there was a misunderstanding. I didn't want to propose to disable the checks, but rather put them at a different place: Instead of checking them with the appimagetool one could add a check in the `project_checks` part of the pipeline. This way, one could use the up-to-date `appstreamcli` to verify the appstream.xml file and keep the CD job at Ubuntu 22.04.

Another option would be to upgrade the Ubuntu Base of the CD job to 24.04., but I guess this would require some additional changes. However, as it will be need at some point, this might also be an option.

**Pull request closed without merging**